### PR TITLE
EKS, ECR, ROUTE53

### PR DESCRIPTION
**Pull request by SBraunovicOmega**

### CODE REVIEW

File: infrastructure/eks.tf [hunk 1]
```diff
@@ -0,0 +1,54 @@
+module "eks" {
+  source  = "terraform-aws-modules/eks/aws"
+  version = "~> 20.0"
+
+  cluster_name                         = "eks-sbraunovic-cluster"
+  cluster_version                      = "1.29"
```
Comment (Igor-Knezevic-VegaIT) [inline infrastructure/eks.tf]: Use latest available version

File: infrastructure/iam.tf [hunk 2]
```diff
@@ -45,6 +45,26 @@ module "iam_iam-policy" {
                 "ecr:DescribeImageScanFindings"
             ],
             "Resource": "*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "eks:AccessKubernetesApi",
```
Comment (Igor-Knezevic-VegaIT) [inline infrastructure/iam.tf]: This is to much, try to follow **principle of least privilege**. 
GitHub should just be able to create eks config and update image tag on the pod level.
Second part is controlled via **access_entries** which you already defined in eks module.

File: infrastructure/irsa.tf [hunk 3]
```diff
@@ -0,0 +1,31 @@
+module "vpc_cni_irsa_role" {
```
Comment (Igor-Knezevic-VegaIT) [inline infrastructure/irsa.tf]: You don't need this for VPC CNI, you will install CNI via **cluster_addons** in eks module.

File: infrastructure/irsa.tf [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+module "irsa_load_balancer" {
+  source    = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+  role_name = "task2-irsa-for-LB"
+
+  attach_load_balancer_controller_policy = true
+
+  oidc_providers = {
+    one = {
+      provider_arn               = module.iam_iam-github-oidc-provider.arn
+      namespace_service_accounts = ["vegait-load-balancer:load-balancer"]
```
Comment (Igor-Knezevic-VegaIT) [inline infrastructure/irsa.tf:10]: make sure you remember namespace and service account names that you put here as they need to be 1 on 1 with the ones that you will put in helm chart when you want to install lb controller.

File: infrastructure/irsa.tf [hunk 5]
```diff
@@ -25,7 +7,7 @@ module "irsa_load_balancer" {
   oidc_providers = {
     one = {
       provider_arn               = module.iam_iam-github-oidc-provider.arn
-      namespace_service_accounts = ["vegait-load-balancer:load-balancer"]
+      namespace_service_accounts = ["${var.namespace_account}:load-balancer"]
```
Comment (Igor-Knezevic-VegaIT) [inline infrastructure/irsa.tf]: this should not be variable, put static namespace name and make sure you put the same in the lb helm chart when you deploy it

Comment (SBraunovicOmega) [reply infrastructure/irsa.tf]: aha, ok :)

File: infrastructure/iam.tf [hunk 6]
```diff
@@ -26,30 +26,44 @@ module "iam_iam-policy" {
...
+        "ecr:DescribeRepositories",
+        "ecr:ListImages",
+        "ecr:DescribeImages",
+        "ecr:BatchGetImage",
+        "ecr:GetLifecyclePolicy",
+        "ecr:GetLifecyclePolicyPreview",
+        "ecr:ListTagsForResource",
+        "ecr:DescribeImageScanFindings"
+      ],
+      "Resource": "*"
```
Comment (Igor-Knezevic-VegaIT) [inline infrastructure/iam.tf]: Make sure you use principle of least privilage.
Check the docs: https://docs.aws.amazon.com/AmazonECR/latest/userguide/image-push-iam.html

**Pull request merged**